client/src/js/components/bhAccountSelect.js

```diff
@@ -44,16 +46,39 @@ function AccountSelectController(Accounts, FormatTreeData, bhConstants, $scope)
     return loadHttpAccounts();
   };
 
+  /**
+   * @function parseAccountTypeIds
+   *
+   * @description
+   * Parses the account type id binding if it is a string or integer and returns an array of
+   * integers.  Also adds in the title account now matter what to the account type array shipped
+   * to the server so that we can always build a tree.
+   */
+  function parseAccountTypeIds(types) {
+    let parsed;
+    if (Array.isArray(types)) {
+      parsed = types.map(type => parseInt(type, 10));
+    } else if (typeof types === 'string') {
+      parsed = types.split(',')
+        .filter(type => type !== '')
+        .map(type => parseInt(type, 10));
+    } else if (typeof types === 'number') {
+      parsed = [types];
+    } else {
+      throw new Error('Cannot parse account types from '.concat(types));
+    }
```

Comment:
I guess we are not waiting to receive an object if it can occur maybe we can handle that also here

Reply:
I actually think it's a good thing to throw an error if the developer doesn't respect the API.  I can't imagine a scenario in which an object should be accepted.